ompi/mca/pml/cm/pml_cm_request.h

```diff
@@ -53,4 +54,22 @@ struct mca_pml_cm_request_t {
 typedef struct mca_pml_cm_request_t mca_pml_cm_request_t;
 OBJ_CLASS_DECLARATION(mca_pml_cm_request_t);
 
+/*
+ * Avoid CUDA convertor inits only for contiguous memory and if indicated by
+ * the MTL. For non-contiguous memory, do not skip CUDA convertor init phases.
+ */
+#if OPAL_CUDA_SUPPORT
+#define MCA_PML_CM_SWITCH_CUDA_CONVERTOR_OFF(flags, datatype, count)            \
+    {                                                                           \
+        if (opal_datatype_is_contiguous_memory_layout(&datatype->super, count)  \
+            && (ompi_mtl->mtl_flags & MCA_MTL_BASE_FLAG_CUDA_INIT_DISABLE)) {   \
+            flags |= CONVERTOR_SKIP_CUDA_INIT;                                  \
+        } else {                                                                \
+            flags &= ~CONVERTOR_SKIP_CUDA_INIT;                                 \
```

Comment:
The flag passed to opal_convertor_copy_and_prepare_for_* is added to the flags of the original convertor (via an or operation). Moreover, it seems you initialize ```flags``` to 0 in all cases, so I don't think there is a need to &~CONVERTOR_SKIP_CUDA_INIT here.